scipy/sparse/coo.py

```diff
@@ -137,16 +137,13 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                 except TypeError:
                     raise TypeError('invalid input format')
 
-                self.row = np.array(ij[0], copy=copy)
-                self.col = np.array(ij[1], copy=copy)
-                self.data = np.array(obj, copy=copy)
-
+                row, col = ij
```

Comment:
Better fix would be to just replace the two `astype` calls with
`self.row = np.asarray(self.row, dtype=idx_dtype)`, as was done in also
some other places.